hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/details/ArtifactDetailsLayout.java

```diff
@@ -264,19 +269,59 @@ private void addGeneratedColumnButton(final Table table) {
             private static final long serialVersionUID = 1L;
 
             @Override
-            public Button generateCell(final Table source, final Object itemId, final Object columnId) {
-                final String fileName = (String) table.getContainerDataSource().getItem(itemId)
-                        .getItemProperty(PROVIDED_FILE_NAME).getValue();
-                final Button deleteIcon = SPUIComponentProvider.getButton(
-                        fileName + "-" + UIComponentIdProvider.UPLOAD_FILE_DELETE_ICON, "",
-                        i18n.getMessage(UIMessageIdProvider.CAPTION_DISCARD), ValoTheme.BUTTON_TINY + " " + "blueicon",
-                        true, FontAwesome.TRASH_O, SPUIButtonStyleNoBorder.class);
-                deleteIcon.setData(itemId);
-                deleteIcon.addClickListener(event -> confirmAndDeleteArtifact((Long) itemId, fileName));
-                return deleteIcon;
+            public HorizontalLayout generateCell(final Table source, final Object itemId, final Object columnId) {
+                HorizontalLayout actioncellLayout = new HorizontalLayout();
+                actioncellLayout.addComponent(getArtifactDeleteButton(table, itemId));
+                actioncellLayout.addComponent(getArtifactDownloadButton(table, itemId));
+                actioncellLayout.setImmediate(true);
+                return actioncellLayout;
             }
         });
+    }
 
+    private Button getArtifactDeleteButton(final Table table, final Object itemId) {
+        final String fileName = (String) table.getContainerDataSource().getItem(itemId)
+                .getItemProperty(PROVIDED_FILE_NAME).getValue();
+        final Button deleteIcon = SPUIComponentProvider.getButton(
+                fileName + "-" + UIComponentIdProvider.UPLOAD_FILE_DELETE_ICON, "",
+                i18n.getMessage(UIMessageIdProvider.CAPTION_DISCARD), ValoTheme.BUTTON_TINY + " " + "blueicon", 
+                true, FontAwesome.TRASH_O, SPUIButtonStyleNoBorder.class);
+        deleteIcon.setData(itemId);
+        deleteIcon.addClickListener(event -> confirmAndDeleteArtifact((Long) itemId, fileName));
+        return deleteIcon;
+    }
+
+    private Button getArtifactDownloadButton(final Table table, final Object itemId) {
+        final String fileName = (String) table.getContainerDataSource().getItem(itemId)
+                .getItemProperty(PROVIDED_FILE_NAME).getValue();
+        final Button downloadIcon = SPUIComponentProvider.getButton(
+                fileName + "-" + UIComponentIdProvider.ARTIFACT_FILE_DOWNLOAD_ICON, "",
+                i18n.getMessage(UIMessageIdProvider.ARTIFACT_DOWNLOAD), ValoTheme.BUTTON_TINY + " " + "blueicon", 
+                true, FontAwesome.DOWNLOAD, SPUIButtonStyleNoBorder.class);
+        downloadIcon.setData(itemId);
+        FileDownloader fileDownloader = new FileDownloader(createStreamResource((Long) itemId));
```

Comment:
could you please still use the confirmation dialog here rather than extending the download icon directly, so you could use previously defined ids and messages

Reply:
Hi @bogdan-bondar - unfortunately, there are no direct ways to bring a customized confirmation dialog along with filedownloader. Please correct me if my understanding is wrong. Anyway, the browser shows a confirmation dialog to open / save the file.

Suggested solution is as below,
	Create an 'invisiblebutton' and extend with filedownloader
	Add an action listener to the download button and invoke 'invisiblebutton.click()' on button event.

So, Shall i proceed with above recommendation? or can we stick to the default confirmation dialog from the browser? Please suggest.

Reply:
Ok, let us rely upon the confirmation dialog coming from browser. Still I would suggest adding the error handler for FileDownloader and removing the stale messages from message.properties as well as BUTTON_YES and BUTTON_NO ids

Reply:
Thanks for the confirmation, I have added error handled for file downloader and also removed unused strings from messages.properties.